### PR TITLE
feat: support `unnest` with additional columns

### LINKED ISSUE #9349: Unnest function doesn't work as expected with additional columns
### Describe the bug

Note that this isn't pertaining to unnest(column1, column2) functionality but rather that what should either work or explicitly fail (because it's not supported) is returning incorrect results.

### To Reproduce

I used the unnest.slt test and added the following:

```
query
select unnest(column1), column3 from unnest_table;
```

If you run 
```
cargo test --test sqllogictests -- unnest --complete
```
this will complete to the following
```
query I
select unnest(column1), column3 from unnest_table;
----
1
2
3
4
5
6
12
```
That result is wrong as it's missing column3 data. 

### Expected behavior

Either an error be returned or the result returns something like 
```
1, 1
2, 1
3, 1
4, 2
5, 2
6, 3
12, null
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonahgao**
## Which issue does this PR close?

Closes #9349.

## Rationale for this change

The `UnnestExec` already has this [functionality](https://github.com/apache/arrow-datafusion/blob/e1ca74e7c68e4949504e902f91d2e2b152cfc910/datafusion/physical-plan/src/unnest.rs#L570).

This PR passes the necessary parameters, i.e., the expressions in the select list that do not need unnesting, to it.

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/sqllogictest/test_files/unnest.slt [hunk 1]
```diff
@@ -91,6 +93,44 @@ NULL
...
+query II
+select unnest(column1), column3 from unnest_table;
+----
+1 1
+2 1
+3 1
+4 2
+5 2
+6 3
+12 NULL
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/unnest.slt:107]: 💯  for testing nulls

File: datafusion/sql/src/select.rs [hunk 2]
```diff
@@ -282,30 +282,38 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
...
-            if exprs_to_unnest.len() > 1 {
+            if unnest_columns.len() > 1 {
                 return not_impl_err!("Only support single unnest expression for now");
             }
-
-            let expr = exprs_to_unnest[0].clone();
-            let column = expr.display_name()?;
-
+            let unnest_column = unnest_columns.pop().unwrap();
+            // Set preserve_nulls to false to ensure compatibility with DuckDB and PostgreSQL
```
Comment (alamb) [inline datafusion/sql/src/select.rs:312]: 👍

**Review by alamb — APPROVED** [review]
Thank you very much for this contribution @jonahgao  -- this PR looks very nice to me (as usual)

**Comment by jonahgao** [comment]
> Thank you very much for this contribution @jonahgao -- this PR looks very nice to me (as usual)

Thank you for the review!

**Review by jayzhan211 — APPROVED** [review]
LGTM Thanks, @jonahgao

**Pull request merged**